### PR TITLE
Restore cpu affinity after loading tensorflow model from transformers

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Describe your changes.
Related with https://github.com/microsoft/onnxruntime/issues/8385

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 1]
```diff
@@ -270,12 +270,17 @@ def load_pt_model(model_name, model_class, cache_dir):
 
 def load_tf_model(model_name, model_class, cache_dir):
     config = AutoConfig.from_pretrained(model_name, cache_dir=cache_dir)
-
+    
+    # Loading tf model from transformers limits the cpu affinity to {0} when KMP_AFFINITY is set
+    # Restore the affinity after model loading for expected ORT performance
+    pid = os.getpid()
+    affinity = os.sched_getaffinity(pid)
```
Comment (mwakaba2) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:277]: FYI, this is only available for linux. This function doesn't exist for Mac.
```python
>>> os.sched_getaffinity(pid)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'os' has no attribute 'sched_getaffinity'
```

Comment (wangyems) [reply onnxruntime/python/tools/transformers/onnx_exporter.py:277]: Oh thanks for the info. seems this is only available on part of Unix. On windows there is another story.

File: onnxruntime/python/tools/transformers/affinity_helper.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        if self.is_os_supported:
+            self.affinity = os.sched_getaffinity(self.pid)
+
+    def set_affinity(self):
+        if self.is_os_supported:
+            current_affinity = os.sched_getaffinity(self.pid)
+            if (self.affinity != current_affinity):
+                logger.warning("Setting new affinity to %s in replace of the current one %s", str(self.affinity),
+                               str(current_affinity))
+            os.sched_setaffinity(self.pid, self.affinity)
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/affinity_helper.py:31]: If self.affinity == current_affinity, you don't need to set, right?

File: onnxruntime/python/tools/transformers/affinity_helper.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    def get_affinity(self):
+        if self.is_os_supported:
+            self.affinity = os.sched_getaffinity(self.pid)
+
+    def set_affinity(self):
+        if self.is_os_supported:
+            current_affinity = os.sched_getaffinity(self.pid)
+            if (self.affinity != current_affinity):
+                logger.warning("Setting new affinity to %s in replace of the current one %s", str(self.affinity),
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/affinity_helper.py:29]: How about "replacing affinity setting %s with %s

File: onnxruntime/python/tools/transformers/affinity_helper.py (hunk 2 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/affinity_helper.py:31]: good catch!

File: onnxruntime/python/tools/transformers/affinity_helper.py (hunk 3 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/affinity_helper.py]: fixed

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**